pkg/apiserver/debugapi/endpoint_def.go

```diff
@@ -0,0 +1,537 @@
+						{Value: "hist-peer"},
+						{Value: "hist-keys"},
+					}),
+					Name: "state", Required: true},
+			},
+			QueryParams: []*endpoint.APIParam{
+				{Model: APIParamModelInt, Name: "bound"},
+			},
+			Middleware: func(ctx *endpoint.Context) {
+				state := ctx.Request.PathValues.Get("state")
+				val := ctx.Request.QueryValues.Get("bound")
+
+				if val == "" {
+					if strings.EqualFold(state, "hist-size") {
+						ctx.Request.QueryValues.Set("bound", "10")
+					} else if strings.EqualFold(state, "hist-keys") {
+						ctx.Request.QueryValues.Set("bound", "10000")
+					}
+				}
+				ctx.Next()
+			},
+		},
+		{
+			ID:        "pd_scheduler_show",
+			Component: model.NodeKindPD,
+			Path:      "/schedulers",
+			Method:    endpoint.MethodGet,
+			QueryParams: []*endpoint.APIParam{
+				{Model: APIParamModelText, Name: "status"},
+			},
+		},
+		{
+			ID:        "pd_stores",
+			Component: model.NodeKindPD,
+			Path:      "/stores",
+			Method:    endpoint.MethodGet,
+			QueryParams: []*endpoint.APIParam{
+				{Model: APIParamModelTags, Name: "state"},
+			},
+			Middleware: func(ctx *endpoint.Context) {
+				vals := ctx.Request.QueryValues["state"]
+				ctx.Request.QueryValues.Del("state")
+				if len(vals) != 0 {
+					for _, state := range vals {
+						stateValue, ok := metapb.StoreState_value[state]
+						if !ok {
+							ctx.Abort(fmt.Errorf("unknown state: %s", state))
+							return
+						}
+						ctx.Request.QueryValues.Add("state", strconv.Itoa(int(stateValue)))
+					}
+				}
+				ctx.Next()
+			},
+		},
+		{
+			ID:        "pd_store_id",
+			Component: model.NodeKindPD,
+			Path:      "/store/{storeID}",
+			Method:    endpoint.MethodGet,
+			PathParams: []*endpoint.APIParam{
+				{Model: APIParamModelText, Name: "storeID", Required: true},
+			},
+		},
+		{
+			ID:        "pd_pprof",
+			Component: model.NodeKindPD,
+			Path:      "/debug/pprof/{kind}",
+			Method:    endpoint.MethodGet,
+			PathParams: []*endpoint.APIParam{
+				pprofKindsParam,
+			},
+			QueryParams: []*endpoint.APIParam{
+				{Model: APIParamModelConstant("1"), Name: "debug"},
+				pprofSecondsParam,
+			},
+			Middleware: func(ctx *endpoint.Context) {
+				if err := timeoutMiddleware(ctx.Request, ctx.Request.QueryValues.Get("seconds")); err != nil {
+					ctx.Abort(err)
+					return
+				}
+				ctx.Next()
+			},
+		},
+		// tikv
+		{
+			ID:        "tikv_config",
+			Component: model.NodeKindTiKV,
+			Path:      "/config",
+			Method:    endpoint.MethodGet,
+		},
+		{
+			ID:        "tikv_profile",
+			Component: model.NodeKindTiKV,
+			Path:      "/debug/pprof/profile",
+			Method:    endpoint.MethodGet,
+			QueryParams: []*endpoint.APIParam{
+				pprofSecondsParam,
+			},
+			Middleware: func(ctx *endpoint.Context) {
+				if err := timeoutMiddleware(ctx.Request, ctx.Request.QueryValues.Get("seconds")); err != nil {
+					ctx.Abort(err)
+					return
+				}
+				ctx.Next(func() error {
+					ctx.Response.Header.Set("Content-Type", "image/svg+xml")
```

Comment:
This is very wired~~ Why isn't it set in the response but we need to manually overwrite?